app/src/lib/stores/app-store.ts

```diff
@@ -2424,13 +2424,21 @@ export class AppStore extends TypedBaseStore<IAppState> {
     const askForConfirmationWhenStashingAllChanges =
       changesState.stashEntry !== null
 
+    let displayViewInGitHub = true
+    if (selectedRepository instanceof Repository) {
+      const hasGitHubRepo = selectedRepository.gitHubRepository !== null
+      const hasRemoteUrl = selectedRepository.url !== null
+      displayViewInGitHub = hasGitHubRepo === hasRemoteUrl
+    }
+
     updatePreferredAppMenuItemLabels({
       ...labels,
       contributionTargetDefaultBranch,
       isForcePushForCurrentRepository,
       isStashedChangesVisible,
       hasCurrentPullRequest: currentPullRequest !== null,
       askForConfirmationWhenStashingAllChanges,
+      displayViewInGitHub,
```

Comment:
I think this `displayViewInGitHub` would be more clearly `isGitHubRepo` and I believe this check could be simplified to `isGitHubRepo: (selectedRepository instanceof Repository) && selectedRepository.gitHubRepository !== null`